packages/astro/src/core/endpoint/index.ts

```diff
@@ -100,12 +97,29 @@ export function createAPIContext({
 	return context;
 }
 
+// Return response only
 export async function callEndpoint<MiddlewareResult = Response | EndpointOutput>(
 	mod: EndpointHandler,
 	env: Environment,
 	ctx: RenderContext,
 	onRequest?: MiddlewareHandler<MiddlewareResult> | undefined
-): Promise<EndpointCallResult> {
+): Promise<Response>;
+// Return response or a simple endpoint object (used for SSG)
+export async function callEndpoint<MiddlewareResult = Response | EndpointOutput>(
+	mod: EndpointHandler,
+	env: Environment,
+	ctx: RenderContext,
+	onRequest?: MiddlewareHandler<MiddlewareResult> | undefined,
+	returnObjectFormIfAvailable?: boolean
+): Promise<Response | SimpleEndpointObject>;
+// Base implementation
+export async function callEndpoint<MiddlewareResult = Response | EndpointOutput>(
+	mod: EndpointHandler,
+	env: Environment,
+	ctx: RenderContext,
+	onRequest?: MiddlewareHandler<MiddlewareResult> | undefined,
+	returnObjectFormIfAvailable?: boolean
+): Promise<Response | SimpleEndpointObject> {
```

Comment:
Had to create this type abomination so we get the correct types depending if `returnObjectFormIfAvailable` is passed.